docs/Operations.md

```diff
@@ -1,8 +1,33 @@
-Operation is a core concept at Maestro, and it represents executions done in multiple layers of Maestro, a state update, or a configuration change. Requests sent from game rooms are not handled in operations. Instead, the services directly take those. In most scenarios, those will be only state updates and won't perform any action on the scheduler or rooms.
+Table of Contents
+---
 
-## Definition and executors
+- [What Is](#what-is)
+  - [Definition and Executors](#definition-and-executors)
+- [How does Maestro handle operations](#how-does-maestro-handle-operations)
+- [States](#state)
+  - [State Machine](#state-machine)
+- [Lifecycle](#lifecycle)
+- [Lease](#lease)
+  - [What is](#what-is-the-operation-lease)
+  - [Why operations have it](#why-operations-have-it)
+  - [Troubleshooting operations with lease](#troubleshooting)
+  - [Operation Lease Lifecycle](#operation-lease-lifecycle)
+- [Available Operations](#available-operations)
 
-Maestro will have multiple operations, and those will be set using pairs of definitions and executors. An operation definition consists of the operation parameters. An operation executor is where the actual operation logic is implemented, and it will receive as input its correlated definition. So, for example, the `CreateSchedulerExecutor` will always receive a `CreateSchedulerDefinition`.
+
+## What is
+Operation is a core concept at Maestro, and it represents executions done in multiple layers of Maestro, a state update, or a configuration change. 
+Requests sent from game rooms are not handled in operations. Instead, the services directly take those. 
+In most scenarios, those will be only state updates and won't perform any action on the scheduler or rooms.
+
+Operations are heavily inspired by the [Command Design Pattern](https://en.wikipedia.org/wiki/Command_pattern)
+
+### Definition and executors
+
+Maestro will have multiple operations, and those will be set using pairs of definitions and executors.
+An operation definition consists of the operation parameters.
+An operation executor is where the actual operation logic is implemented, and it will receive as input its correlated definition.
```

Comment:
```suggestion
An operation executor is where the actual operation execution and rollback logic is implemented, and it will receive as input its correlated definition.
```